actors/builtin/miner/partition_state.go

```diff
@@ -158,20 +157,83 @@ func (p *Partition) AddFaults(store adt.Store, sectorNos *abi.BitField, sectors
 	return power, nil
 }
 
-// Removes sector numbers from faults and thus from recoveries.
-// The sectors are removed from the Faults and Recovering bitfields, and FaultyPower and RecoveringPower reduced.
-// The sectors are re-scheduled for expiration shortly after their target expiration epoch.
-// Consistency between the partition totals and queue depend on the reported sectors actually being faulty and recovering.
-// Returns the power of the now-recovered sectors.
-func (p *Partition) RecoverFaults(store adt.Store, recovered *abi.BitField, sectors []*SectorOnChainInfo,
-	ssize abi.SectorSize, quant QuantSpec) (PowerPair, error) {
+// Declares a set of sectors faulty. Already faulty sectors are ignored,
+// terminated sectors are skipped, and recovering sectors are reverted to
+// faulty.
+//
+// - New faults are added to the Faults bitfield and the FaultyPower is increased.
+// - The sectors' expirations are rescheduled to the fault expiration epoch, as "early" (if not expiring earlier).
+//
+// Returns the power of the now-faulty sectors.
+func (p *Partition) DeclareFaults(
+	store adt.Store, sectors Sectors, sectorNos *abi.BitField, faultExpirationEpoch abi.ChainEpoch,
+	ssize abi.SectorSize, quant QuantSpec,
+) (newFaults *bitfield.BitField, newFaultyPower PowerPair, err error) {
+	err = validateFRDeclarationPartition(p, sectorNos)
```

Comment:
Could we rename this to `validatePartitionContainsSectors` since that's what it's doing? `validateSectorDeclaration` or `validateFaultDeclaration` would also be more meaningful and accurate, while permitting additional validations.